From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jonathan Schwender <schwenderjonathan@gmail.com>
Date: Mon, 27 Jul 2026 19:21:17 +0100
Subject: [PATCH] libz: Use include path from libz-sys

The vendored libz-sys folder was never included in the
cargo package, so the previous include can't have worked.
It probably wasn't a problem in practice, since libz include
headers are usually installed.
Nevertheless this should be fixed, and the usual way is using the
`DEP_` syntax to read the `include` variable printed from the build-script
of libz-sys, which gives us the header file locations.
---
 Cargo.toml |  4 ++--
 build.rs   | 17 +++++++++++++++--
 libz-sys   |  1 -
 3 files changed, 17 insertions(+), 5 deletions(-)
 delete mode 160000 libz-sys

diff --git a/Cargo.toml b/Cargo.toml
index 587b318..456ad71 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,10 +14,10 @@ homepage = "https://github.com/PistonDevelopers/freetype-sys"
 edition = "2018"
 
 [features]
-bundled = []
+bundled = ["libz-sys"]
 
 [dependencies]
-libz-sys = "1.1.20"
+libz-sys = {  version = "1.1.20", optional = true }
 
 [build-dependencies]
 cc = "1"
diff --git a/build.rs b/build.rs
index 3ed8e8a..04f00c6 100644
--- a/build.rs
+++ b/build.rs
@@ -1,4 +1,5 @@
 use std::env;
+use std::env::VarError;
 
 fn add_sources(build: &mut cc::Build, root: &str, files: &[&str]) {
     let root = std::path::Path::new(root);
@@ -92,10 +93,22 @@ fn main() {
 
     build.compile("freetype2");
 
-    println!("cargo:rustc-link-lib=z");
+    // libz-sys comma separates multiple include paths.
+    let zlib_include_paths = match env::var("DEP_Z_INCLUDE") {
+        Ok(include_paths) => include_paths.split(",").map(String::from).collect(),
+        Err(VarError::NotPresent) => {
+            // For some targets (e.g. android and -ohos), libz-sys does not emit the variable,
+            // but we expect the header to be in the sysroot instead so it's fine.
+            vec![]
+        }
+        Err(VarError::NotUnicode(_os_str_paths)) => {
+            println!("cargo:warning:libz-sys header files at non-unicode path. Ignoring");
+            vec![]
+        }
+    };
 
     let mut build = cc::Build::new();
-    build.include("libpng").include("libz-sys/src/zlib");
+    build.include("libpng").includes(zlib_include_paths);
     build
         .file("libpng/png.c")
         .file("libpng/pngerror.c")
diff --git a/libz-sys b/libz-sys
deleted file mode 160000
index 1742670..0000000
--- a/libz-sys
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit 17426707016e4052653a06a908c5f193553b7d6f
